compiler/src/yul/selectors.rs

```diff
@@ -0,0 +1,175 @@
+use crate::errors::CompileError;
```

Comment:
something that came across my mind when reviewing this was the genericity of the function names `switch`, `expression` and `case`. you could make the argument that we rebind the names on import elsewhere and they make sense in the context of this module, but it is something to consider.

i left a comment at a usage of `switch` to point out what I mean

Reply:
I agree. There must be applicable CS nomenclature for this.

Reply:
i'd just go with what sounds natural and readable. and ideally unique across as many contexts as possible. of course, we quickly end up optimizing a many-dimensional space and can't satisfy all of our constraints at once :)

off the top of my head, i would call the thing `expression` as `predicate` as the `selector::predicate`, while still jargon-y is clearer to me which part of the expression it is referring to, rather than just `selector::expression`.

and i'd prolly add some more semantic content to the symbol for `case` -- something like `switch_case` or `arm` or something.... this being said if these functions don't leak outside of this module there is less need to do this given the context gained as a local reader of this module